lib/usecase/submit_survey_use_case.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:injectable/injectable.dart';
+import 'package:survey_flutter_ic/api/exception/network_exceptions.dart';
+import 'package:survey_flutter_ic/api/repository/survey_repository.dart';
+import 'package:survey_flutter_ic/api/request/submit_survey_request.dart';
+import 'package:survey_flutter_ic/usecase/base/base_use_case.dart';
+
+@Injectable()
+class SubmitSurveyUseCase extends UseCase<void, SubmitSurveyRequest> {
```

Comment:
@Wadeewee The `SubmitSurveyRequest` should be used in Repository or Service layer instead. In the UseCase, we should have a new `Input` model for the UseCase's param

Reply:
@hoangnguyen92dn, May I know why we should not map `model` to `request` since in `VM` 🙏

Reply:
> Technically, the request/response models are supposed to interact with the backend side which contains `toJson` or `fromJson` for API usage, in the MVVM Clean architecture, the Repository layer is in charge of handling data with data source (Remote or Local),

I agreed with this 🤝  @hoangnguyen92dn